File: Python3/프로그래머스/2/42578. 의상/의상.py
import math

def solution(clothes):
    clothes_dict = { }
    
    for v, k in clothes:
        if k in clothes_dict.keys():
            clothes_dict[k] += 1
            
        else:
            clothes_dict[k] = 1
        
    for k in clothes_dict.keys():
        clothes_dict[k] += 1
    
    return math.prod(clothes_dict.values()) - 1

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    unordered_map<string, int> um;
    
    for(const vector<string>& s: clothes) {
        um[s[1]]++;
    }
    
    for(const auto& [key, value]: um) {
        answer *= (value + 1);
    }
    
    return answer - 1;
}